src/js/hyperschedule.ts

```diff
@@ -854,15 +897,86 @@ function processSearchText() {
       }, filterKeywordsValues)
     ) {
       filtersText.push(text);
+    } else if (isTimeRange(text)) {
+      // Looks only at one time range.
+      timeText = text;
     } else {
       queryText.push(text);
     }
   }
 
   const query = getSearchQuery(queryText);
   const filters = getSearchTextFilters(filtersText);
+  const time = getTimeFilter(timeText);
+  return { query, filters, time };
+}
+
+function isTimeRange(searchText: string) {
+  const timeArray = searchText.split(":");
+  // Time can be singular or a range
+  // Ex: 3:00PM or 3:00PM-4:00PM
+  return (
+    (timeArray.length === 3 &&
+      !isNaN(+timeArray[0]) &&
+      timeArray[1].includes("-")) ||
+    (timeArray.length === 2 &&
+      !isNaN(+timeArray[0].charAt(timeArray[0].length - 1)) &&
+      !isNaN(+timeArray[1].substring(0, 2)))
+  );
+}
```

Comment:
This stuff seems kind of convoluted.  In fact, it doesn't appear to be used by `getTimeRange` or `getTimeFilter`, so is it needed at all?

Reply:
This is currently used by `processSearchText()` to see if a search query is indeed a specified time/time range. I the `!isNan` was an interesting way we found to check if a string is numeric, but I believe this can be done better through a regex.